dali/kernels/signal/fft/fft_postprocess.cuh

```diff
@@ -110,38 +110,58 @@ struct power_dB {
   }
 };
 
+
 template <typename Out, typename In, typename Convert>
 __global__ void ConvertTimeMajorSpectrogram(
       Out *out, int out_stride,
       const In *in, int in_stride, int nfft, int64_t nwindows,
       Convert convert = {}) {
+  int64_t wnd = static_cast<int64_t>(blockIdx.x) * blockDim.y + threadIdx.y;
+  if (wnd >= nwindows)
+    return;
+  out += wnd * out_stride;
+  in += wnd * in_stride;
+
+  for (int i = threadIdx.x; i < nfft; i += 32)
+    out[i] = convert(in[i]);
+}
+
+template <typename Out, typename In, typename Convert>
+__global__ void ConvertTimeMajorSpectrogram_InPlaceDiffTypeSize(
```

Comment:
Until I found that we are only using it with float and float2 it looked a bit like UB.